lib/puma/request.rb

```diff
@@ -266,14 +271,12 @@ def normalize_env(env, client)
         uri = URI.parse(env[REQUEST_URI])
         env[REQUEST_PATH] = uri.path
 
-        raise "No REQUEST PATH" unless env[REQUEST_PATH]
```

Comment:
Actually, the Rack spec says:

```
PATH_INFO: This may be an empty string, if the request URL targets the application root and does not have a trailing slash.
```

So this is also a rack conformance issue

Reply:
Rack spec specifies the PATH_INFO and REQUEST_SPEC seems to be a non-spec addition (to quote you) https://github.com/puma/puma/issues/1441#issuecomment-774784265. Should puma fail when it's nil? Maybe we could just assign an empty PATH_INFO then (as I do below)?

Reply:
It looks like for HTTP URIs it will always return at least an empty string, but if someone gives us a weird protocol maybe it could be nil.

I think nil.to_s is a reasonable conversion.

Reply:
I was testing it with `config.ru` 
```
require "localhost"
run proc { [200, {"Content-Type" => "text/plain"}, ["Hello, World!"]] }
```
and a request
```
 wget https://github.com/ -e https_proxy=http://localhost:9292
```

With this request REQUEST_PATH was `nil`.